Tests/Microsoft.AppCenter.Test.Windows/Storage/StorageAdapterTest.cs

```diff
@@ -26,218 +23,211 @@ public class StorageAdapterTest
+                // Db file might not exist or might fail to be deleted.
             }
+        }
+
+        /// <summary>
+        /// Verify that database file is created when Initialize() is called.
+        /// </summary>
+        [TestMethod]
+        public void CreateDbDiskImageOnInitialization()
+        {
+            InitializeStorageAdapter();
+        }
+
+        /// <summary>
+        /// Verify that Initialize would fail with incorrect db path passed.
+        /// </summary>
+        [TestMethod]
+        public void FailOnOpenDatabaseWithWrongName()
+        {
             try
             {
-                // Try get data before database initialize.
-                CreateTableHelper(adapter);
+                _adapter.Initialize("test://test.txt");
             }
             catch (Exception e)
             {
-                Assert.Fail("Shouldn't have thrown exception");
+                Assert.IsTrue(e.Message.Contains("Failed to open database connection"));
             }
-            adapter.Dispose();
+        }
+
+        /// <summary>
+        /// Verify that storage adapter is disposed when calling Dispose().
+        /// </summary>
+        [TestMethod]
+        public void StorageAdapterDisposedOnDisposeCall()
+        {
+            InitializeStorageAdapter();
+            _adapter.Dispose();
+            var exceptionThrown = false;
             try
             {
-                // Try get data before database initialize.
-                CreateTableHelper(_adapter);
-                Assert.Fail("Should have thrown exception");
+                CreateTable();
             }
-            catch (Exception e)
+            catch
             {
-                Assert.IsFalse(e.Message.Contains("Should have thrown exception"));
+                exceptionThrown = true;
             }
+            Assert.IsTrue(exceptionThrown);
         }
 
         /// <summary>
-        /// Verify that database is not initilaze.
+        /// Verify that database is not initialized.
         /// </summary>
         [TestMethod]
-        public void DatabaseIsNotInitilazeWhenCallCount()
+        public void DatabaseIsNotInitializedWhenCallCount()
         {
             // Prepare data.
-            var exception = new StorageException("The database wasn't initialized.");
+            var exceptionThrown = false;
             try
             {
-                // Try get data before database initialize.
+                // Try to get data before database initialization.
                 _adapter.Count(TableName, ColumnChannelName, StorageTestChannelName);
-                Assert.Fail("Should have thrown exception");
             }
             catch (Exception e)
             {
-                Assert.AreEqual(exception.Message, e.Message);
+                exceptionThrown = true;
+                Assert.AreEqual("The database wasn't initialized.", e.Message);
             }
+            Assert.IsTrue(exceptionThrown);
```

Comment:
You can simplify the code by moving the check inside the catch block, and then no need for exceptionThrown variable.